server/pbench/bin/pbench-server-prep-shim-001

```diff
@@ -0,0 +1,220 @@
+
+typeset -i ntbs=0
+typeset -i nerrs=0
+typeset -i nquarantined=0
+typeset -i ndups=0
+
+echo $TS
+while read tblink ;do
+    # resolve the link
+    tb=$(readlink -f ${tblink})
+    sts=$?
+    if [ $sts != 0 ] ;then
+        echo "$TS: readlink failed: ${tblink}" | tee -a $status >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    # directory
+    tbdir=$(dirname ${tb})
+
+    # resultname: get the basename foo.tar.xz and then strip the .tar.xz
+    resultname=$(basename ${tb})
+    resultname=${resultname%.tar.xz}
+
+    # hostname is the last component of the directory part of the full path
+    controller=$(basename ${tbdir})
+
+    # TBD
+    dest=$archive/$controller
+
+    # sanity checks
+    # Impossible: the tarball disappeared between the find and this check!
+    if [ ! -f ${tb} ] ;then
+        echo "$TS: ${tb} file does not exist" | tee -a $status >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    if [ ! -f ${tb}.md5 ] ;then
+        # this may be the result of pbench-move-results racing with this script:
+        # we'll just skip the tarball this time and try again next time.
+        echo "$TS: ${tb}.md5 file does not exist" >&4
+        continue
+    fi
+
+    # This should not happen: version 001 agents check for duplicates in pbench-move-results.
+    if [ -f ${dest}/${resultname}.tar.xz -o -f ${dest}/${resultname}.tar.xz.md5 ] ;then
+        quarantine $duplicates/$controller  ${tb} ${tb}.md5
+        sts=$?
+        if [ $sts -ne 0 ] ;then
+            echo "Failed: quarantine $quarantine/$controller ${tb}, status $sts" |
+                tee -a $status >&4
+        fi
+        ndups=$ndups+1
+    fi
+
+    pushd ${tbdir} >/dev/null 2>&1
+    # MD5 check
+    md5sum --check $(basename $tb).md5 
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: ${tb} failed MD5 check" >&4
+        quarantine $quarantine/$controller ${tb} ${tb}.md5
+        sts=$?
+        if [ $sts -ne 0 ] ;then
+            echo "Failed: quarantine $quarantine/$controller ${tb} ${tb}.md5, status $sts" |
+                tee -a $status >&4
+        fi
+        nquarantined=$nquarantined+1
+        nerrs=$nerrs+1
+        popd >/dev/null 2>&4
+        continue
+    fi
+
+    # if necessary, fix prefix file name.
+    prefix=""
+    if [ -f "prefix.${resultname}" ] ;then
+       # rename it - using a prefix form was a mistake.
+        mv prefix.${resultname} ${resultname}.prefix
+    fi
+    # ... and remember it
+    if [ -f "${resultname}.prefix" ] ;then
+        prefix="${tbdir}/${resultname}.prefix"
+    fi
+    popd >/dev/null 2>&1
+
+    # make the destination directory if necessary.
+    mkdir -p ${dest}
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: Failed: \"mkdir -p ${dest}\", status $sts" |
+            tee -a $status >&4
+        nerrs=$nerrs+1
+        # this is fatal
+        break
+    fi
+
+    mv ${tb} ${tb}.md5 ${prefix} $dest
+    sts=$?
+    if [ $sts -ne 0 ] ;then
```

Comment:
Can we quarantine when this `mv` command fails?

Reply:
Absolutely.